sdk/python/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb

```diff
@@ -244,6 +251,41 @@
     "In order to use this data to create an AzureML MLTable, we first need to convert it to the required JSONL format. The following script is creating two `.jsonl` files (one for training and one for validation) in the corresponding MLTable folder. The train / validation ratio corresponds to 20% of the data going into the validation file. For further details on jsonl file used for image classification task in automated ml, please refer to the [data schema documentation for multi-class image classification task](https://learn.microsoft.com/en-us/azure/machine-learning/reference-automl-images-schema#image-classification-binarymulti-class)."
    ]
   },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## First generate the jsonl file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "\n",
+    "sys.path.insert(0, \"../jsonl-conversion/\")\n",
```

Comment:
Can we use relative import instead of adding this to sys.path here ?

Reply:
After trying several things and reading lots of forums, I don't think this is possible from a notebook.

https://stackoverflow.com/questions/14132789/relative-imports-for-the-billionth-time
The well accepted answer on this post states that if the __name__ variable is __main__ such as when running a script from terminal or using the interactive terminal, relative imports won't work, as they are mainly for packages. I printed the __name__ variable in the notebook and it is __main__ which I believe is why I keep getting `ImportError: attempted relative import with no known parent package` despite adding __init__.py files. 

The relative import even fails if I try to import from the same directory (e.g. `from . import MyModule`). Some more searching leads me to believe that using sys package is common practice for notebooks to get around this issue.

Have you been successful in using relative imports in notebooks before?

Reply:
Thanks for the detailed analysis and explanation. I didn't know that relative imports wouldn't work in a notebook. Will close this comment.